sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -867,8 +870,22 @@ def _delete_dataset(self, project_id, dataset_id, delete_contents=True):
   @retry.with_exponential_backoff(
       num_retries=MAX_RETRIES,
       retry_filter=retry.retry_on_server_errors_and_timeout_filter)
-  def create_temporary_dataset(self, project_id):
+  def create_temporary_dataset(self, project_id, source_table_reference=None):
     dataset_id = BigQueryWrapper.TEMP_DATASET + self._temporary_table_suffix
+    location = None
+
+    # If given, use source table location as location for the temp dataset
+    tr = source_table_reference
+    if tr is not None:
+      if tr.projectId is None:
+        # if the source table has no projectId, assume the given project_id
+        source_project_id = project_id
```

Comment:
You should use the executing project here (project_id). The entity executing the query might not have permissions to create a Dataset in the project that owns the source table (but new Dataset should be in the same region as the source Dataset).